src/libraries/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/AesImplementation.cs

```diff
@@ -43,6 +43,50 @@ protected sealed override void Dispose(bool disposing)
             base.Dispose(disposing);
         }
 
+        protected override bool TryDecryptEcbCore(
+            ReadOnlySpan<byte> ciphertext,
+            Span<byte> destination,
+            PaddingMode paddingMode,
+            out int bytesWritten)
+        {
+            UniversalCryptoTransform transform = CreateTransformCore(
```

Comment:
In my head, the implementation of these was going to be a very simple saved key object with a very simple just-call-the-OS body.

Looking at how things work (the protected KeyValue field, OS key handles having the (algorithm, key, mode) triplet, not just the (algorithm, key) doublet), that doesn't seem.... sane.

If creating this UniversalCryptoTransform really becomes a crisis of allocation (which I doubt) we can pursue complexity later... but we're probably fine.

Simple complexity option 1: override the Key property, make the setter clear out all allocated algorithm handles, call a GetAlgorithmHandle thingy to get an eventually consistent handle... oh, but then we lose thread-safety.  Was thread-safe (except changing the key) a goal?  Maybe. I don't remember.

Simple complexity option 2: We can probably structify UniveralCryptoTransform? But we still have the SafeHandles, so, it's probably never worth doing.

Reply:
I agree. I implemented and threw away this work twice, one of which was "simple call in to the OS"

> but we're probably fine.

Yeah. And it can be done as follow up work if necessary. Right now I am conscious of it also being May.